editoast/src/views/search.rs

```diff
@@ -353,13 +352,22 @@ pub async fn search(
 
     if dry {
         let query = diesel::debug_query::<Pg, _>(&query).to_string();
-        return Ok(HttpResponse::Ok().body(query));
+        return Ok(axum::response::Response::builder()
+            .status(StatusCode::OK)
+            .body(axum::body::Body::from(query))
+            .unwrap());
     }
 
     let conn = &mut db_pool.get().await?;
     let objects = query.load::<SearchDBResult>(conn).await?;
     let results: Vec<_> = objects.into_iter().map(|r| r.result).collect();
-    Ok(HttpResponse::Ok().json(results))
+    Ok(axum::response::Response::builder()
+        .status(StatusCode::OK)
+        .header(CONTENT_TYPE, "application/json")
+        .body(axum::body::Body::from(
+            serde_json::to_string(&results).expect("serde error"),
+        ))
+        .expect("response builder error"))
```

Comment:
Can't you do something like this instead?

```suggestion
    Ok((StatusCode::Ok, Json(serde_json::to_string(&results).expect("serde error")))

Reply:
Sadly we can't because it's not the same type as the other `return` of the function : https://github.com/OpenRailAssociation/osrd/pull/8047/files/035803e3552389598686696a86e2892775cbcc4a..488e4c81dc2ff3140643a0e863293890b1ef0a8d#diff-b6a2310b76e9bd3bab165e74f48fbf50e3619c5d178c7c2bcfef596e2845f434R355-R358

That other branch used to return a `content-type: text`. But it sure we can return a JSON string instead.

Reply:
I'd say it's weird that the other return doesn't return the same type so I'd say maybe we need to address that in the first place. Returning the text as a JSON is probably a good idea?

Reply:
I've added to the list of #8178. This way we can make it a good first issue (we need these 😬).